packages/manager/apps/telecom/src/app/telecom/pack/migration/pack-migration.controller.js

```diff
@@ -29,7 +41,43 @@ export default class TelecomPackMigrationCtrl {
           this.process.migrationTaskId = head(pendingTasks);
           this.process.migrationDoing = true;
         } else {
-          this.process.currentStep = 'offers';
+          this.getPackAccess().then((packServices) => {
+            const [service] = packServices;
+            if (service.includes('xdsl')) {
+              // Retrieve line associated to the service
+              this.getLine(service).then((line) => {
+                // Search building by line to retrieve buildings elements
+                this.searchBuildingByLine(line).then((buildings) => {
```

Comment:
Please avoid Promise calls imbrication

Reply:
I would like but I don't know how to do it.